x/gov/handler.go

```diff
@@ -595,22 +598,36 @@ func (h TextResolutionHandler) Check(ctx weave.Context, db weave.KVStore, tx wea
 }
 
 func (h TextResolutionHandler) Deliver(ctx weave.Context, db weave.KVStore, tx weave.Tx) (*weave.DeliverResult, error) {
-	_, err := h.validate(ctx, db, tx)
+	msg, err := h.validate(ctx, db, tx)
 	if err != nil {
 		return nil, err
 	}
-	// TODO: store this resolution somewhere
-	return &weave.DeliverResult{}, nil
+
+	proposal := CtxProposal(ctx)
+	if proposal == nil {
+		return nil, errors.Wrap(errors.ErrNotFound, "no proposal set for passed resolution")
+	}
+	resolution := &Resolution{
+		Metadata: msg.Metadata,
+		// Note: right here is where I wish to have ID in the model
```

Comment:
@ethanfrey @husio why don't we introduce a proposalID into the context as part of this PR?
I see no problem introducing a `withProposalID` method which is what I will do and then we can discuss alternatives.
If we want to store the ID - we have to do it now instead of creating a ticket for later?

Reply:
@husio this has been done in the last commit, I have introduced a wrapper for propsal to save and retrieve it with an ID. I think it looks acceptable.

Reply:
I think using context to pass temporary variables is a misuse of context. I would question existing context getter/setter if I undestand the code better :see_no_evil: 


I am sharing opinion with what is described [in this blog post](https://peter.bourgon.org/blog/2016/07/11/context.html) - temporary variables should not be inserted into the context.

Reply:
Well, we just added an id there, I'd like to postpone the discussion about the general approach of storing a proposal in the context to some other time as this has already been done before and this PR just enhances that with an ID. But I get where you are coming from.

Reply:
@husio I have read this blog post.

And I have worked with a number of go frameworks that did add important info to the context (including stuff like db references). We pass a number of tx-specific variables there, which are considered optional by most of the code, and the essential ones (db, tx) are passed explicitly.

While I agree with you in theory, I have often struggled to find any good generic way to do without them. (And large nesting on anonymous functions only works if they are defined in one place and is h**l for code-reuse in my experience)

Reply:
@ruseinov I am fine with adding a second context ProposalID along with Proposal. And we clean it up later.

Let's get this working now, I will prepare some more PRs